tests/build-ai-analyze.test.ts

```
@@ -0,0 +1,126 @@
+
+beforeEach(() => {
+  mockSupabaseApikey.mockReset()
+  mockCheckPermission.mockReset()
+  mockGetEnv.mockReset()
+  mockGetEnv.mockImplementation((_: unknown, key: string) => {
+    if (key === 'BUILDER_URL') return builderUrl
+    if (key === 'BUILDER_API_KEY') return builderApiKey
+    return ''
+  })
+  // @ts-expect-error global fetch mock
+  global.fetch = vi.fn()
```

ESLint

[error] Unexpected use of 'global'. Use `globalThis` instead.

(no-restricted-globals)